src/core/highlight-vars.js

```diff
@@ -0,0 +1,106 @@
+/**
+ * Module core/highlight-vars
+ * Highlights occurrences of a <var> within a section on click.
+ * Set `conf.highlightVars = true` to enable.
+ * Removes highlights from <var> if clicked anywhere else.
+ * All is done while keeping in mind that exported html stays clean
+ * on export.
+ */
+import { sub } from "core/pubsubhub";
+import hlVars from "deps/text!core/css/var.css";
+
+export const name = "core/highlight-vars";
+
+export function run(conf) {
+  if (!conf.highlightVars) {
+    return;
+  }
+  const styleElement = document.createElement("style");
+  styleElement.textContent = hlVars;
+  styleElement.classList.add("removeOnSave");
+  document.head.appendChild(styleElement);
+
+  [...document.querySelectorAll("var")].forEach(varElem =>
+    varElem.addEventListener("click", hightlightListener)
+  );
+
+  // remove highlights, cleanup empty class/style attributes
+  sub("beforesave", outputDoc => {
+    [...outputDoc.querySelectorAll("var.respec-hl")].forEach(removeHighlight);
+  });
+}
+
+function hightlightListener(ev) {
+  ev.stopPropagation();
+  const { target: varElem } = ev;
+  const hightligtedElems = highlightVars(varElem);
+  const resetListener = () => {
+    const hlColor = getHighlightColor(varElem);
+    hightligtedElems.forEach(el => removeHighlight(el, hlColor));
+    [...HL_COLORS.keys()].forEach(key => HL_COLORS.set(key, true));
+  };
+  if (hightligtedElems.length) {
+    document.body.addEventListener("click", resetListener, { once: true });
+  }
+}
+
+// availability of highlight colors. colors from var.css
+const HL_COLORS = new Map([
```

Comment:
TODO: in the future, we should make these into configurable themes and "levels". So, instead of this, we just say `${conf.highlight.theme}-level-${X}`. 

Then we provide a bunch of themes that people may like. Let's see how popular this gets first.

Reply:
I din't get what you mean by "level". Can you please clarify?